_pages/assignments.html

```diff
@@ -28,9 +28,11 @@ <h1>Assignments</h1>
                         <p>
                             <a href="{{ site.data.constants.broadway_on_demand_link }}" class="btn btn-success" role="button">Broadway On-Demand Autograder</a>
                         </p>
-                        <p>
-                            <a href="{{ site.data.constants.malloc_contest_link }}" class="btn btn-warning" role="button">Malloc Contest</a>
-                        </p>
+                        {% for mp in site.data.assignments['mps'] %}{% if mp.name != "Malloc" or mp.visible != true %} {% continue %} {% endif %}
+                            <p>
+                                <a href="{{ site.data.constants.malloc_contest_link }}" class="btn btn-warning" role="button">Malloc Contest</a>
+                            </p>
+                        {% endfor %}
```

Comment:
Roughly translating this:
```
for mp in mps {
    if mp.name != "Malloc" or mp.visible != true {
        continue
    }
    display malloc contest button
}
```

I think a more intuition code flow would be:
```
for mp in mps {
    if mp.name == "Malloc" and mp.visible == true {
        display malloc contest button
        break
    }
}
```

Reply:
Is `== true` needed for this templating engine? Just asking because in general it is a code smell.

Reply:
I think jinja is used to inject pythonic code into html. So no I do not think `== true` is required. Will edit my suggestion.

Reply:
![image](https://user-images.githubusercontent.com/31669118/73596707-2dd21480-44ea-11ea-8694-e843cc8f4267.png)
Sounds good, I was originally trying to copy the style found a little lower in the file, although it might be more helpful to make this easier to read. I'll change it